pylinalg/matrix.py

```diff
@@ -631,6 +631,242 @@ def mat_look_at(eye, target, up_reference, /, *, out=None, dtype=None):
     return out
 
 
+def _mat_inv(m):
+    # Reference:
+    # https://github.com/mrdoob/three.js/blob/dev/src/math/Matrix4.js
+    # based on:
+    # http://www.euclideanspace.com/maths/algebra/matrix/functions/inverse/fourD/index.htm
+    out = np.empty((4, 4), dtype=m.dtype)
```

Comment:
I think the return type of np.linalg.inv is always float, so it's fine to do that here as well.